screens/Menu/Settings.js

```diff
@@ -15,21 +32,128 @@ import PropTypes from "prop-types";
-     */
+    //Put function in Async storage
+    const getSwitchValue = async () => {
+        let getAccessToken = await AsyncStorage.getItem("accessToken");
+        let switchValue = await AsyncStorage.getItem("switchVal");
+        if (switchValue == "true") {
+            const resp = await getUsersCalendarList(getAccessToken);
+            getFilteredGoogleCalendarList(resp);
+        }
+        setSwitchVal2(switchValue);
+        setAccessToken(getAccessToken);
+    };
+
+    const signInOrOut = async (val) => {
+        if (val) {
+            const respCalendars = await signInWithGoogleAsync();
+            getFilteredGoogleCalendarList(respCalendars);
+        }
+        else {
+            if (accessToken) {
+                /* Log-Out */
+                setCalendarList(null);
+                await Google.logOutAsync({
+                    accessToken, iosClientId: "128383090622-lgrk639fn4k6t99lhrldkh02441fcjgb.apps.googleusercontent.com",
+                });
+            }
+        }
+    }
+
+    async function signInWithGoogleAsync() {
+        try {
+            const result = await Google.logInAsync({
+                // androidClientId: YOUR_CLIENT_ID_HERE,
+                iosClientId: "128383090622-lgrk639fn4k6t99lhrldkh02441fcjgb.apps.googleusercontent.com",
+                scopes: ['profile', 'email', 'https://www.googleapis.com/auth/calendar.readonly'],
+            });
+            if (result.type === 'success') {
+                AsyncStorage.setItem("accessToken", result.accessToken);
+                setAccessToken(result.accessToken);
+                return getUsersCalendarList(result.accessToken); //Here We are getting all the calendars in a json...
+            } else {
+                return { cancelled: true };
+            }
+        } catch (e) {
+            return { error: true };
+        }
+    }
+
+    const getFilteredGoogleCalendarList = (respCalendars) => {
+        var filteredList = respCalendars.items.map(element => {
+            return { id: element.id, summary: element.summary, description: element.description };
+        });
+        setCalendarList(filteredList);
+    };
+
+    const getUsersCalendarList = async (accessToken) => {
+        let calendarsList = await fetch('https://www.googleapis.com/calendar/v3/users/me/calendarList', {
+            headers: { Authorization: `Bearer ${accessToken}` },
+        });
+        let resp = await calendarsList.json();
+        return resp;
+    }
+
+    const press = (item) => {   // The onPress method 
+        var { checked } = isConnected;
+        // These ensures that multiple checkboxes don't all get affected when one is clicked
+        if (!checked.includes(item) || checked.length == 1) {
+            checked = [];
+            setIsConnected({ checked: [...checked, item] });
+        } else {
+            setIsConnected({ checked: checked.filter(a => a != item) });
+        }
+    };
+
+
+    // //CHECK IF IT WORKS
```

Comment:
What is going on here? We should reduce commented code if we can.

Reply:
I was just trying to create the method for reducing the size of the labels. It is not working i forgot to remove it.